guides/md/keras-tuner/distributed_tuning.md

```diff
@@ -0,0 +1,170 @@
+
+<div class="k-default-codeblock">
+```
+["python: can't open file 'run_tuning.py': [Errno 2] No such file or directory"]
+
+```
+</div>
+Example bash script for worker:
+
+
+```python
+!export KERASTUNER_TUNER_ID="tuner0"
+!export KERASTUNER_ORACLE_IP="127.0.0.1"
+!export KERASTUNER_ORACLE_PORT="8000"
+!python run_tuning.py
+```
+
+<div class="k-default-codeblock">
+```
+python: can't open file 'run_tuning.py': [Errno 2] No such file or directory
+
+```
+</div>
+### Data parallelism with tf.distribute
+
+Keras Tuner also supports data parallelism via [tf.distribute](https://www.tensorflow.org/tutorials/distribute/keras). Data parallelism and distributed tuning can be combined. For example, if you have 10 workers with 4 GPUs on each worker, you can run 10 parallel trials with each trial training on 4 GPUs by using [tf.distribute.MirroredStrategy](https://www.tensorflow.org/api_docs/python/tf/distribute/MirroredStrategy). You can also run each trial on TPUs via [tf.distribute.experimental.TPUStrategy](https://www.tensorflow.org/api_docs/python/tf/distribute/experimental/TPUStrategy). Currently [tf.distribute.MultiWorkerMirroredStrategy](https://www.tensorflow.org/api_docs/python/tf/distribute/experimental/MultiWorkerMirroredStrategy) is not supported, but support for this is on the roadmap.
+
+
+### Example code
+
+When the enviroment variables described above are set, the example below will run distributed tuning and will also use data parallelism within each trial via `tf.distribute`. The example loads MNIST from `tensorflow_datasets` and uses hyperband for the hyperparameter search.
+
+
+```python
+
+import kerastuner as kt
+import tensorflow as tf
+import numpy as np
+
+
+def build_model(hp):
+    """Builds a convolutional model."""
+    inputs = tf.keras.Input(shape=(28, 28, 1))
+    x = inputs
+    for i in range(hp.Int("conv_layers", 1, 3, default=3)):
+        x = tf.keras.layers.Conv2D(
+            filters=hp.Int("filters_" + str(i), 4, 32, step=4, default=8),
+            kernel_size=hp.Int("kernel_size_" + str(i), 3, 5),
+            activation="relu",
+            padding="same",
+        )(x)
+
+        if hp.Choice("pooling" + str(i), ["max", "avg"]) == "max":
+            x = tf.keras.layers.MaxPooling2D()(x)
+        else:
+            x = tf.keras.layers.AveragePooling2D()(x)
+
+        x = tf.keras.layers.BatchNormalization()(x)
+        x = tf.keras.layers.ReLU()(x)
+
+    if hp.Choice("global_pooling", ["max", "avg"]) == "max":
+        x = tf.keras.layers.GlobalMaxPooling2D()(x)
+    else:
+        x = tf.keras.layers.GlobalAveragePooling2D()(x)
+    outputs = tf.keras.layers.Dense(10, activation="softmax")(x)
+
+    model = tf.keras.Model(inputs, outputs)
+
+    optimizer = hp.Choice("optimizer", ["adam", "sgd"])
+    model.compile(
+        optimizer, loss="sparse_categorical_crossentropy", metrics=["accuracy"]
+    )
+    return model
+
+
+def main():
+    """Runs the hyperparameter search."""
+    tuner = kt.Hyperband(
```

Comment:
Can you double check if we should call a tuner with `kt.tuners.Hyperband` instead of `kt.Hyperband`?

e.g. `from kerastuner.tuners import Hyperband` or `RandomSearch` (https://keras-team.github.io/keras-tuner/examples/helloworld/)

It seems to work in both instances but the latter may be the correct way? (Will  update this here too https://www.tensorflow.org/tutorials/keras/keras_tuner)

Reply:
Yes, the `kt.tuner.Hyperband` is importing it from the source file, which is close to where the implementation is. The suggested way of using it would be `kt.Hyperband`.

@fchollet Do you have any comment on this?

Reply:
Thanks for the reply @haifeng-jin ! Can we suggest to users that if they want to use built-in tuners, they can use either way?

For example, in this blog post - https://blog.tensorflow.org/2020/01/hyperparameter-tuning-with-keras-tuner.html :

> Keras Tuner comes with Bayesian Optimization, Hyperband, and Random Search algorithms built-in...
>
> ```
> tuner = kt.tuners.BayesianOptimization(...)
> ```
> ...
> ```
> tuner = kt.Hyperband(...)
> ```

Reply:
We'd generally recommend using top-level symbols (like `kt.Hyperband`), referring to the submodules is pure cognitive overhead for users.

Reply:
Thank you @fchollet @haifeng-jin 👍 

@MarkDaoust This also means we can close a PR in /docs that proposes `kt.tuners.HyperBand` since it's not a fix (both `kt.tuners.HyperBand` and `kt.HyperBand` work though).